From c703bb82ed5ff477eefd97391990da46c3e782c4 Mon Sep 17 00:00:00 2001
From: Huang Chenyi <b44402@freescale.com>
Date: Thu, 13 Jun 2013 17:20:27 +0800
Subject: [PATCH 15/39] ENGR00266008 3G Icon still show after connect Net
 cable

Set DataTypeIconId and QSDataTypeIconId to zero
when 3G is not connected.

Signed-off-by: Huang Chenyi <b44402@freescale.com>
---
 .../src/com/android/systemui/statusbar/policy/NetworkController.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
index 09f1695..d931f9a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
@@ -685,6 +685,11 @@ public class NetworkController extends BroadcastReceiver implements DemoMode {
             }
         }
 
+        if(!mDataConnected){
+           mDataTypeIconId = 0;
+           mQSDataTypeIconId = 0;
+        }        
+
         if (isCdma()) {
             if (isCdmaEri()) {
                 mDataTypeIconId = R.drawable.stat_sys_data_fully_connected_roam;
-- 
1.8.0

